scripts/windows/msys2_prepare.bat

```diff
@@ -66,74 +143,41 @@ mingw-w64-x86_64-python3-pip ^
 mingw-w64-x86_64-python3-pynacl ^
 mingw-w64-x86_64-python3-yaml
 
-echo Current time: %time%
-
-rem Find and execute the VS env preparation script
-FOR /F "usebackq delims=" %%F IN (`where /f /r "C:\Program Files (x86)\Microsoft Visual Studio" vcvarsall`) DO SET vcvarsall=%%F
-call %vcvarsall% %PLATFORM%
-
-rem Set the paths appropriately
-SET "PATH=%WIN_MSYS2_ROOT%\mingw64\bin;%WIN_MSYS2_ROOT%\usr\bin;%WIN_OTP_PATH%\bin;%WIN_OTP_PATH%\erts-%OTP_VERSION%;%PATH%"
+@call:log Set up msys2 env variables
+:: 1>nul to hide the Copied 1 file message
+COPY %~dp0msys2_env_build.sh %WIN_MSYS2_ROOT%\etc\profile.d\env_build.sh 1>nul
 
-echo Current time: %time%
-rem Set up msys2 env variables
+SET BASH="%WIN_MSYS2_ROOT%\usr\bin\bash.exe"
```

Comment:
Unfortunately setting Path is not always enough to guarantee that we execute the proper bash (i.e. we can have WSL, or git bash already installed). So we better implicitly refer to the executable.